src/Microsoft.TestPlatform.Client/TestPlatform.cs

```diff
@@ -92,6 +92,16 @@ public IDiscoveryRequest CreateDiscoveryRequest(DiscoveryCriteria discoveryCrite
             // Update cache with Extension Folder's files
             this.AddExtensionAssemblies(discoveryCriteria.RunSettings);
 
+            // Update and initialize loggers only when DesignMode is false
+            var runConfiguration = XmlRunSettingsUtilities.GetRunConfigurationNode(discoveryCriteria.RunSettings);
```

Comment:
RunConfiguration is heavily used. Should we consider making `RunConfiguration` as property in `RunSettingsManager` instead of creating an object of RunConfiguration every time we need to read run config?

Reply:
cc : @codito